From a43546100759a0545a52d6df4fa6cdd270e701af Mon Sep 17 00:00:00 2001
From: CI Assistant <chdauto@intel.com>
Date: Sat, 11 Jan 2020 12:06:15 +0200
Subject: [PATCH] Merge pull request #1634 in SW_UGW/linux from
 bugfix/UGW_SW-44499-setting-upper-limit-for-trying-queue-empty-841 to 8.4.1

* commit '2de61840433d941ea8a6cbf7df7d02a0b25de537':
  UGW_SW-44499: Adding max limit while trying to empty queue
---
 drivers/net/ethernet/lantiq/cqm/grx500/cbm.c | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
index 5e61f2315f42..49609729e35d 100644
--- a/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
+++ b/drivers/net/ethernet/lantiq/cqm/grx500/cbm.c
@@ -3468,14 +3468,17 @@ static s32 cbm_empty_queue(s32 cbm_port_id, u32 qocc, u32 port_type)
 	while (qocc) {
 		#ifdef CHECK_WHILE_LOOP
 		iter++;
-		if (iter > 30000) {
-			pr_info_once("%s >30k iter qocc %d", __func__, qocc);
+		if (iter > 50000) {
+			pr_info_once("%s >50k iter qocc %d\n", __func__, qocc);
+			return qocc;
+		} else if (iter > 30000) {
+			pr_info_once("%s >30k iter qocc %d\n", __func__, qocc);
 		} else if (iter > 10000) {
-			pr_info_once("%s >10k iter qocc %d", __func__, qocc);
+			pr_info_once("%s >10k iter qocc %d\n", __func__, qocc);
 		} else if (iter > 5000) {
-			pr_info_once("%s >5k iter qocc %d", __func__, qocc);
+			pr_info_once("%s >5k iter qocc %d\n", __func__, qocc);
 		} else if (iter > 500) {
-			pr_info_once("%s >500 iter qocc %d", __func__, qocc);
+			pr_info_once("%s >500 iter qocc %d\n", __func__, qocc);
 		}
 		#endif
 		for (i = 0; i < dqm_port_info[cbm_port_id].deq_info.num_desc;) {
